docs: replace images with text in mode-of-operation (fixes #31)

MChandrahas opened the pull request:
### Description
This PR addresses issue #31 by replacing the images in `docs/explanation/mode-of-operation.md` with text descriptions.

### Changes
- Removed the introductory paragraph ("The purpose of Chisel is...") as requested in the issue.
- Removed the HTML table and images.
- Converted the visual workflow steps into a numbered list with descriptive text.


### QA Steps
1. Open `docs/explanation/mode-of-operation.md`.
2. Verify that the introductory paragraph is removed.
3. Verify that the images are replaced by the 4-step text workflow.

MChandrahas committed:
docs: replace images with text in mode-of-operation (fixes #31)

MChandrahas committed:
Merge branch 'main' into fix/31-replace-images-mode-of-operation

asanvaq reviewed (changes requested):
Thank you for your contribution! We have been discussing this change among the team, and we reached the conclusion that a more visual alternative to the images is needed (eg. a diagram). We want to replicate the original version without using any images. I understand this adds a layer of complexity to this issue, so please let me know if you would like to continue working on this!

MChandrahas committed:
docs: replace images with mermaid diagrams

MChandrahas committed:
Merge branch 'fix/31-replace-images-mode-of-operation' of https://github.com/MChandrahas/chisel-docs into fix/31-replace-images-mode-of-operation

asanvaq commented:
Thanks for the update! The Read the docs project is not building, I suspect it's because the “sphinxcontrib-mermaid” extension is not added to requirements.txt and config.py. Instructions on how to do this can be found [here](https://canonical-starter-pack.readthedocs-hosted.com/stable/how-to/diagrams-as-code-mermaid/#installation-and-setup).

MChandrahas committed:
build: fix mermaid diagrams rendering

MChandrahas commented:
Fixed! I added the missing "sphinxcontrib-mermaid" dependency and configured _myst_fence_as_directive_ in conf.py. The local build is passing now without warnings

MChandrahas committed:
docs: remove obsolete image files

MChandrahas committed:
Fix mermaid diagram height issue causing large gaps

MChandrahas commented:
## Fix: Mermaid diagram height issue

### Problem
The mermaid extension generates inline `<style>` with a fixed height of 500px, causing large gaps above and below diagrams.

### Solution
Added `custom.js` with a MutationObserver that watches for DOM changes and sets `svg.style.height = 'auto'` dynamically.

MChandrahas committed:
Merge branch 'main' into fix/31-replace-images-mode-of-operation

asanvaq commented:
Yes! It sounds good :)

asanvaq commented:
Closed due to inactivity.
